src/components/TabBar/TabBar.tsx

```diff
@@ -1,30 +1,46 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
+import { useQuery } from "../../hooks/useQuery";
 import Tab from "./Tab/Tab";
 import styles from "./TabBar.module.scss";
 
 interface TabBarProps {
-  handleTabClick: (tab: string)=>{}
+  handleTabClick: (tab: string) => {};
+  // activeTab: string;
 }
 
-const TabBar: React.FC<TabBarProps> = ({handleTabClick}) => {
+const TabBar: React.FC<TabBarProps> = ({ handleTabClick }) => {
   const tabsArr = ["Popular", "Top rated", "Upcoming"];
+  const tabs: {[key: string]: string} = {
+    popular : "Popular",
+    top_rated : "Top rated",
+    upcoming : "Upcoming",
+  }
+  const queryTab = useQuery('filter');
 
-  const [activeTab, setActiveTab] = useState(tabsArr[0]);
+  const [activeTab, setActiveTab] = useState('');
 
   const handleClick = (tab: string): void => {
-    setActiveTab(tab);
     handleTabClick(tab);
-  }
+  };
+
+  useEffect(()=>{
+    if (queryTab !== null){
+      setActiveTab(tabs[queryTab])
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [queryTab])
 
   return (
-    <nav className={styles["tabs-wrapper"]}>
-      {tabsArr.map((tab, index) => {
-        return tab === activeTab ? (
-          <Tab key={index} label={tab} active handleClick={handleClick} />
-        ) : (
-          <Tab key={index} label={tab} handleClick={handleClick} />
-        );
-      })}
+    <nav>
+      <ul className={styles["tabs-wrapper"]}>
+        {tabsArr.map((tab, index) => {
+          return tab === activeTab ? (
+            <Tab key={index} label={tab} active handleClick={handleClick} />
+          ) : (
+            <Tab key={index} label={tab} handleClick={handleClick} />
```

Comment:
let's combine in one 
`<Tab key={index} label={tab} handleClick={handleClick}  active={tab === activeTab}/>`